test/template-tests/testcases/CreateUIDefinition/VMSizes-Must-Match-Template.test.ps1

```diff
@@ -0,0 +1,28 @@
+﻿param(
+[Parameter(Mandatory=$true)]
+[PSObject]
+$CreateUIDefinitionObject,
+
+[Parameter(Mandatory=$true)]
+[Collections.IDictionary]
+$MainTemplateParameters
+)
+
+$sizeSelectors = $CreateUIDefinitionObject | 
+    Find-AzureRMTemplate -Key type -Value Microsoft.Compute.SizeSelector
+
+
+foreach ($selector in $sizeSelectors) {
+    $MainTemplateParam = $MainTemplateParameters[$selector.Name]

```

Comment:
is this asserting that the name of the parameter in mainTemplate must equal the name of the control in CUID?  If so, that's not nec true - we'd need to match these via outputs in CUID

Reply:
Please show me the pairing you'd prefer.

Reply:
The param name in mainTemplate has no relation to that in CUID - the mapping is done in the output section.  We can't assume the names are the same but will have to parse this way:

basics('controlName') and
steps('stepName').controlName

and then map that to the property name in the outputs section.  So you could have:

outputs:
"vmPassword": "[basics('nameOfPasswordControl').password]"
or
"vmPassword": "[steps('stepName').nameOfPasswordControl.password]"